daemon: Use DeepCopy method from StatusResponse

vadorovsky opened the pull request:
This change replaces usage of mohae/deepcopy with a generated DeepCopy
method of StatusResponse. mohae/deepcopy did not handle copying
strfmt.DateTime value properly.

Fixes: a9e759f ("daemon: do deep copy of model.StatusResponse")
Fixes: #6605

Signed-off-by: Michal Rostecki <mrostecki@suse.de>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/6650)
<!-- Reviewable:end -->

vadorovsky committed:
daemon: Use DeepCopy method from StatusResponse

This change replaces usage of mohae/deepcopy with a generated DeepCopy
method of StatusResponse. mohae/deepcopy did not handle copying
strfmt.DateTime value properly.

Fixes: a9e759f477d1 ("daemon: do deep copy of model.StatusResponse")
Fixes: #6605

Signed-off-by: Michal Rostecki <mrostecki@suse.de>

vadorovsky committed:
dep: vendor: Update go-openapi/strfmt and remove mohae/deepcopy

Update go-openapi/strfmt to include DeepCopy and DeepCopyInto methods in
all types. [0]

Remove mohae/deepcopy which is not used anymore and does not work
correct for all types.

[0] https://github.com/go-openapi/strfmt/commit/dd2bd34dedb77725bec3b8e941ad06daf32a935a

Signed-off-by: Michal Rostecki <mrostecki@suse.de>

vadorovsky commented on vendor/github.com/go-openapi/strfmt/zz_generated.deepcopy.go:
```diff
@@ -0,0 +1,35 @@
+// +build !ignore_autogenerated
```
I'm open for suggestions where should I add this file. Or for general suggestions how we should add DeepCopy method to strfmt.

I was thinking about having a fork of strfmt, but that might be too much effort for just one file.

tgraf replied:
Can we submit it upstream?

vadorovsky replied:
Good idea, done https://github.com/go-openapi/strfmt/pull/44

vadorovsky replied:
Upstream PR merged. I'm going to bump go-openapi version and update the PR.

tgraf replied:
Awesome!

aanm reviewed (changes requested):
When I first looked at this PR I got curious how you did solve the DeepCopy for DateTime, now I know

@mrostecki the reason I'm not ok with this change is because that we are modifying the vendor directory. All code in the vendor directory should be the same of the commit specified for that vendor in `Gopkg.toml`. There's no easy way to solve this problem AFAIK.

vadorovsky committed:
api: Generate DeepCopy methods for StatusResponse

This change adds autogenerated DeepCopy and DeepCopyInto methods for
StatusResponse and all its children.

Signed-off-by: Michal Rostecki <mrostecki@suse.de>
